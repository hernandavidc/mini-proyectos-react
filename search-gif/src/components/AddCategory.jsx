import { useState } from "react";

export const AddCategory = (props) => {

  const [inputValue, setInputValue] = useState('');

  const onInputChange = ({ target }) => setInputValue(target.value);

  const onSubmit = (event) => {
      event.preventDefault();
      if(inputValue.trim().length <= 1) return;

      setInputValue('');
      props.onAddCategory(inputValue.trim());
  };

  return (
    <form onSubmit={ onSubmit } >
        <input
            type="text"
            placeholder="Buscar gifs"
            value={ inputValue }
            onChange={ (e) => onInputChange(e, props) }
        />
    </form>
  )
}
