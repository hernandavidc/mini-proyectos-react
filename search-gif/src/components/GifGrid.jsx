import { useState, useEffect } from "react";
import { GifItem } from "./GifItem";

const getGifs = async (category) => {
  const url = `https://api.giphy.com/v1/gifs/search?q=${category}&limit=10&api_key=d5YoY0PK8cJr7yDrgvlyhGP1lpJfs9ur`;
  const { data } = await fetch(url).then((res) => res.json());

  const temp_gifs = data.map( ({ id, title, images }) => ({
     id,
     title,
     url: images.downsized_medium.url
  }));
  return temp_gifs;
}

export const GifGrid = ({ category }) => {

  const [gifs, setGifs] = useState([ ]);

  useEffect(() => {
    getGifs(category).then( gifs => setGifs(gifs) );
  }, [ ])

  return ( <>
    <section className="card-grid">
      <h3> { category } </h3>
      { gifs.map( (item) => (
        <GifItem key={ item.id }
          { ...item }
        />
      )) }
    </section>
  </> )
}