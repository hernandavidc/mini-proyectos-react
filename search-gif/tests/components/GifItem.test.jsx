import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en <GifItem />', () => {
  const title = 'doggie';
  const url = 'https://i.giphy.com/media/8H80IVPjAdKY8/200w.webp';

  test('debe de hacer match con el snapshot', () => {
    const { container } = render( <GifItem title={ title } url={ url } /> );

    expect(container).toMatchSnapshot();
  });

  test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
    render( <GifItem title={ title } url={ url } /> );
    // screen.debug()
    // expect(screen.getByRole('img').src).toBe(url);
    // expect(screen.getByRole('img').alt).toBe(title);

    const { src, alt} = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('Debe de mostrar el titulo en el componente', () => {
    render( <GifItem title={ title } url={ url } /> );

    expect(screen.getByText(title)).toBeTruthy();
  });
});