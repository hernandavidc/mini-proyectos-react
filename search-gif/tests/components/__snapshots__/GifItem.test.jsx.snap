// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="doggie"
      src="https://i.giphy.com/media/8H80IVPjAdKY8/200w.webp"
    />
    <p>
      doggie
    </p>
  </div>
</div>
`;
